### DISCUSSION TITLE
Temporarily disable spot requests globally

### DISCUSSION THREAD

**Discussion by TimPetricola** [runs-on/runs-on#49] [Ideas]
We sometimes have spikes of jobs, resulting in _a lot_ of spot instances requested. Even though we have a high service quota, we sometimes exceed it, which results in a lot of `MaxSpotInstanceCountExceeded` EC2 API errors. As each job tried to get a spot instance, fails, and has to fallkback on requesting an on-demand instance, it increase the queuing time significantly.

Would it be possible to temporarily disable spot instance requests in such cases (without changing the workflow definitions)? 

I'm thinking about either:
- a parameter on AppRunner to disable it
- do it automatically (e.g. RunsOn gets a first `MaxSpotInstanceCountExceeded`, it disabled spot requests for 1min)

Thanks!

**Comment by crohr** [comment]
Just for reference, this would be the error message received when attempting the launch:

```
[{"ErrorCode":"MaxSpotInstanceCountExceeded","ErrorMessage":"Max spot
instance count
exceeded"
```

**Reply by ri-roee** [reply]
We just ran into this same issue - requested an increase in vCPUs globally on spot instances but this would be a great failsafe.

**Reply by crohr** [reply]
Totally doable, but I think I still need to send an alert so that you know that you have to up your quota

**Reply by crohr** [reply]
This should land in https://github.com/runs-on/runs-on/pull/56

**Comment by crohr** [comment]
Implemented in v2.2.0. Hard to test though, so will see how it behaves in the real-world.
